File: src/main/java/br/tech/gtech/spring/standardizedresponses/core/service/ResponseService.java
package br.tech.gtech.spring.standardizedresponses.core.service;

import org.springframework.stereotype.Service;

import br.tech.gtech.spring.standardizedresponses.core.model.CommonResult;
import br.tech.gtech.spring.standardizedresponses.core.model.SingleResult;

@Service
public class ResponseService {
    public <T> CommonResult getResult(T data) {
        return getSingleResult(data);
    }

    private <T> CommonResult getSingleResult(T data) {
        SingleResult<T> result = new SingleResult<>();
        result.setData(data);
        return result;
    }
}

File: bin/src/main/java/br/tech/gtech/spring/core/service/ResponseService.kt
package br.tech.gtech.spring.core.service;

import br.tech.gtech.spring.core.model.SingleResult
import org.springframework.stereotype.Service;

@Service
public class ResponseService {
    public <T> CommonResult getResult(T data) {
        return getSingleResult(data);
    }

    private <T> CommonResult getSingleResult(T data) {
        SingleResult<T> result = new SingleResult<>();
        result.setData(data);
        return result;
    }
}